src/rprblender/operators/export_training_data.py

```diff
@@ -0,0 +1,174 @@
+#**********************************************************************
+# Copyright 2020 Advanced Micro Devices, Inc
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#********************************************************************
+"""
+Render scene and save predefined AOVs at a specific sample
+"""
+
+from pathlib import Path
+from bpy.props import BoolProperty, IntProperty, StringProperty
+
+import pyrpr
+from rprblender.export import object, camera
+from rprblender.engine.export_engine import ExportEngine, ExportEngine2
+
+from . import RPR_Operator
+
+from rprblender.utils.logging import Log
+
+log = Log(tag='operators.export_training_data')
+
+
+class RPR_EXPORT_OP_export_training_data(RPR_Operator):
+    bl_idname = "rpr.export_training_data"
+    bl_label = "Export render training data"
+    bl_description = "Export training data for an active scene. " \
+                     "Render from each camera and save the result: " \
+                     "color, albedo, depth, shading_normal aovs to the " \
+                     "file formats bin, png, exr, etc"
+
+    output_path: StringProperty(
+        default='',
+        name='Output path',
+        subtype= "DIR_PATH",
+        description="Set directory for output files"
+    )
+    use_scene_resolution: BoolProperty(
+        default=True,
+        name="Use Scene Resolution",
+        description="If True uses scene render resolution, else check image_width and image_height"
+    )
+    extension: StringProperty(
+        default="png",
+        name="Extension",
+        description="Filename extension for output file: png, bin, jpg, exr, tif, etc"
+    )
+    width: IntProperty(
+        default=1920,
+        name="Width",
+        description="Set render width resolution",
+    )
+    height: IntProperty(
+        default=1080,
+        name="Height",
+        description="Set render height resolution",
+    )
+    samples: StringProperty(
```

Comment:
bpy.props.IntVectorProperty technically I'd rather have this be an int vector property.  In all honesty though I think this can be hard coded as this is the only set the denoiser needs.

Reply:
Ok, got it, will change the set.

Reply:
We discussed this option. IntVectorProperty is size fixed array. I suggested to make it as StringProperty, becuase it gives us more flexibility in setting samples including testing.
@bsavery if you think better to use IntVectorProperty or hardcode this, let it be.